desktop-exporter/server.go

```diff
@@ -51,27 +51,30 @@ func tracesHandler(store *TraceStore) func(http.ResponseWriter, *http.Request) {
 	}
 }
 
-func getTraceIDHandler(traceStore *TraceStore) func(http.ResponseWriter, *http.Request) {
+func traceIDHandler(store *TraceStore) func(http.ResponseWriter, *http.Request) {
 	return func(writer http.ResponseWriter, request *http.Request) {
-		traceStore.mut.Lock()
-		defer traceStore.mut.Unlock()
-
 		traceID := mux.Vars(request)["id"]
-		jsonTrace, err := json.Marshal(traceStore.traceMap[traceID])
+		traceData, err := store.GetTrace(traceID)
 		if err != nil {
 			fmt.Println(err)
+			writer.WriteHeader(http.StatusInternalServerError)
```

Comment:
Something like this maybe?

```go
traceData, err := store.GetTrace(traceID)
if err != nil {
  fmt.Println(err)
  writer.WriteHeader(http.StatusInternalServerError)
  return
}

jsonTraceData, err := json.Marshal(traceData)
if err != nil {
  fmt.Println(err)
  writer.WriteHeader(http.StatusInternalServerError)
  return
}

writer.WriteHeader(http.StatusOK)
writer.Header().Set("Content-Type", "application/json")
writer.Write(jsonTraceData)
```

Reply:
Thank you this is exactly what I was looking for! 

![axolotl](https://user-images.githubusercontent.com/56372758/201767134-6068d8a9-6677-4e74-a6c9-f9623078347a.gif)